src/app/common/components/MustGatherBtn.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { WarningTriangleIcon } from '@patternfly/react-icons';
+import { MustGatherContext } from '@app/common/context';
+import { getMustGatherApiUrl } from '@app/queries/helpers';
+import { IMustGatherResponse } from '@app/client/types';
+import { UseQueryResult } from 'react-query';
+import { IPlan } from '@app/queries/types';
+
+interface IMustGatherBtn {
+  mg: IMustGatherResponse | undefined;
+  mustGathers: UseQueryResult<IMustGatherResponse[], Response>;
+  customName: string;
+  type: 'plan' | 'vm';
+}
+
+export const MustGatherBtn: React.FunctionComponent<IMustGatherBtn> = ({
+  mg,
+  mustGathers,
+  customName,
+  type,
+}) => {
+  const {
+    setMustGatherModalOpen,
+    setActiveMustGather,
+    mustGatherWatchList,
+    setMustGatherWatchList,
+  } = React.useContext(MustGatherContext);
+
+  return (
+    <>
+      {mg?.status === 'completed' && mg?.['archive-name'] ? (
+        <Tooltip
+          content={
+            !mustGathers.isSuccess
+              ? `Could not reach must gather service.`
+              : `${mg?.['archive-name']} available for download`
+          }
+        >
+          <Button
+            isAriaDisabled={!mustGathers.isSuccess}
+            variant="secondary"
+            component={'a'}
+            target="_blank"
+            download
+            href={getMustGatherApiUrl(`must-gather/${mg?.['id']}/data`)}
+          >
+            Download logs
+          </Button>
+        </Tooltip>
+      ) : (
+        <>
+          <Tooltip
+            content={
+              !mustGathers.isSuccess
+                ? `Could not reach must gather service.`
+                : mg?.status === 'inprogress'
+                ? `Collecting ${type === 'plan' ? 'migration plan' : 'VM migration'} logs.`
+                : mg?.status === 'new'
+                ? `Must gather queued for execution.`
+                : mg?.status === 'error'
+                ? `Could not complete must gather for ${mg['custom-name']}`
+                : `Collects the current ${
+                    type === 'plan' ? 'migration plan' : 'VM migration'
+                  } logs and creates a tar archive file for download.`
+            }
+          >
+            <Button
+              icon={mg?.status === 'error' ? <WarningTriangleIcon /> : null}
+              isLoading={
```

Comment:
I really like this new `isLoading` prop of button, there are definitely other places in our UI where we just disable a button and show a spinner below it or whatever. Pretty much everywhere we use `<ResolvedQuery />` to handle a mutation... that would be neat to clean up later.